taxonomy/algolia/utils.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+"""
+Utility functions related to algolia indexing.
+"""
+import logging
+
+from django.conf import settings
+
+from taxonomy.algolia.client import AlgoliaClient
+from taxonomy.algolia.constants import ALGOLIA_INDEX_SETTINGS
+from taxonomy.algolia.serializers import JobSerializer
+from taxonomy.models import Job
+
+LOGGER = logging.getLogger(__name__)
+
+
+def index_jobs_data_in_algolia():
+    """
+    Re-Index all jobs data to algolia.
+
+    This function is responsible for
+        1. Constructing a list of dicts containing all jobs present in the database.
+        2. Re-Indexing all data in a single atomic operations with zero downtime.
+
+    Note: We need to construct a list of all jobs in the form of a list and the send it all in a single attempt to
+        make the operation atomic and make sure there is no downtime. Paginating DB data and incrementally adding
+        objects to thew index will cause downtime equal to the amount of time it will take to run the command.
```

Comment:
is there an upper limit on how much data can be sent in a single go? what will happen if data is exceeds the limit?

Reply:
@muhammad-ammar Internally, algolia client sends the records in a batches of 1000 records each. So there is no limit from algolia perspective. 

the only limit here is getting the records from db and storing them in memory (in a list) But I doubt we would have that many jobs that it will be an issue.